src/main/ruby/core/truffle/exception_operations.rb

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+module Truffle
+  module ExceptionOperations
+    def self.class_name(receiver)
+      Truffle::Type.object_class(receiver).name
+    end
+
+    NO_METHOD_ERROR = Proc.new do |exception|
+      if exception.receiver.respond_to?(:inspect)
+        format('undefined method `%s\' for %s:%s', exception.name, exception.receiver, class_name(exception.receiver))
```

Comment:
`inspect` is not called here, but `to_s` is called instead, isn't it?

Reply:
Good spot, and your suggested test restructure caught it.